sklearn/neural_network/tests/test_mlp.py

```diff
@@ -662,6 +662,17 @@ def test_warm_start():
             clf.fit(X, y_i)
 
 
+def test_warm_start_full_iteration():
```

Comment:
We should parametrize the test with pytest and check that we have the same behaviour with the regressor and classifier

```suggestion
@pytest.mark.parametrize("MLPEstimator", [MLPClassifier, MLPRegressor])
def test_warm_start_full_iteration(MLPEstimator):
    # Non-regression test for:
    # https://github.com/scikit-learn/scikit-learn/issues/16812
    # Check that the MLP estimator only accomplish `max_iter` with a
    # warm started estimator.
```

Reply:
Good point, didn't think of that. I will look into it and update the PR (but leaving for a weekend of mountaineering soon)

Reply:
Done that with the last set of commits pushed